saleor/product/utils/variants.py

```diff
@@ -69,6 +70,7 @@ def fetch_variants_for_promotion_rules(
                 for variant_id in set(variants.values_list("pk", flat=True))
             ]
         )
+        product_ids += variants.values_list("product", flat=True)
```

Comment:
product_ids += variants.values_list("product_id", flat=True)

Reply:
that is exactly the same query no mater if you use `product_id` or just `product`. Is this just a convention to use `product_id` or something else?

Reply:
I was afraid that using `product` causes fetching product instances. `product_id` explicitly says that we only want the `products_id` which is a column on the variant table. If `variants.values_list("product", flat=True)` is doing the same underhood, it's ok